Formula/a/asn.rb

```diff
@@ -0,0 +1,42 @@
+class Asn < Formula
+  desc "Organization lookup and server tool (ASN / IPv4 / IPv6 / Prefix / AS Path)"
+  homepage "https://github.com/nitefood/asn"
+  url "https://github.com/nitefood/asn/archive/refs/tags/v0.74.tar.gz"
+  sha256 "d51895526b1a98ae6865094aac10c1abb05b515005702f5a23ac8e1c60e159e8"
+  license "MIT"
+  head "https://github.com/nitefood/asn.git", branch: "master"
+
+  depends_on "aha"
+  depends_on "bash"
+  depends_on "coreutils"
+  depends_on "grepcidr"
+  depends_on "ipcalc"
+  depends_on "jq"
+  depends_on "mtr"
+  depends_on "nmap"
+
+  uses_from_macos "curl"
+  uses_from_macos "whois"
+
+  def install
+    bin.install "asn"
+  end
+
+  def caveats
+    <<~EOS
+      To enable the "AS path tracing" feature, MTR should be in $PATH.
+      If a simple brew link doesn't work:
+        brew link mtr
+      One option could be to manually add a symbolic link to the MTR executable:
+        sudo ln -sfn /usr/local/sbin/mtr /usr/local/bin/mtr
+        sudo ln -sfn /usr/local/sbin/mtr-packet /usr/local/bin/mtr-packet
+      Another option could be to have in $PATH the dir:
+        /usr/local/sbin
+    EOS
+  end
+
+  test do
+    output = shell_output("#{bin}/asn -h 2>&1 | grep --line-buffered -E '\\s+\\d+\\.\\d+' | head -n 1")
+    assert_match version.to_s, output.strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Test logic updated running a simple example